contents/ruby/Rails.md

```diff
@@ -0,0 +1,157 @@
+    // Call to an addEntry function in controller
+  });
+
+  app.get('/name', (req, res, next) => {
+    // Call to a fetchAll function in controller
+  });
+
+  app.get('/name/:id', (req, res, next) => {
+    // Call to a fetch function in controller
+  });
+
+  app.put('/name/update/:id', (req, res, next) => {
+    // Call to an update function in controller 
+  });
+
+  app.delete('/name/delete/:id', (req, res, next) => {
+    // Call to a delete function in controller
+  });
+
+```
+
+Each one of these routes had to be written manually, and specific routes had to be deliberated upon, so as to not be confusing or break any established conventions the application is already following. Also, if you open the source linked at the top, you might notice that the actual code is far more bloated than this abbreviated version, and involves much more logic. Now, to contrast this with the Rails approach, you first need to declare a resource, namely, an `entry` in the address book. So, in your `routes.rb` file, you would write something like
+
+```ruby
+resources :entries
+```
+
+Now, you can check all the routes that you have in your application by running `rails routes`. This should include:
+
+```
+                   Prefix Verb   URI Pattern                                                                              Controller#Action
+                  entries GET    /entries(.:format)                                                                       entries#index
+                          POST   /entries(.:format)                                                                       entries#create
+                new_entry GET    /entries/new(.:format)                                                                   entries#new
+               edit_entry GET    /entries/:id/edit(.:format)                                                              entries#edit
+                    entry GET    /entries/:id(.:format)                                                                   entries#show
+                          PATCH  /entries/:id(.:format)                                                                   entries#update
+                          PUT    /entries/:id(.:format)                                                                   entries#update
+                          DELETE /entries/:id(.:format)                                                                   entries#destroy
+``` 
+
+The word 'automagically' should finally start to make sense at this point! Instead of going through the arduous process of writing out the code dump at the top, and deciding the specific names of the routes, as well as the corresponding functions in the controller, Rails has already configured all the URI patterns with function names in the controller for entries. Now all you need to do is focus on writing the function logic, instead of coding out the names of the individual URIs and routes.
```

Comment:
What a less subjective alternative to this paragraph might look like:
> Resource routing in Rails allows you to quickly declare all of the common routes for a given resource in your application. Instead of defining separate URI patterns for each controller action (which, by convention, maps to a specific CRUD operation in your database), a resourceful route declares them in a single line of code.

And then somewhere around here you can include the link to the relevant Rails guide:
https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default

Reply:
I thought it's okay to be subjective, since this part of the 'Why' and Prof's comment said that I can give my opinions here. I don't mind switching to your version, but that is a lot more business-y and factual. What do you say?